packages/transport-native-ble/src/api/bleApi.ts

```diff
@@ -0,0 +1,127 @@
+/* eslint-disable require-await */
+import { Device as BleDevice } from 'react-native-ble-plx';
+
+import {
+    AbstractApi,
+    AbstractApiConstructorParams,
+    DEVICE_TYPE,
+} from '@trezor/transport/src/api/abstract';
+import * as ERRORS from '@trezor/transport/src/errors';
+import { AsyncResultWithTypedError } from '@trezor/transport/src/types';
+
+import { log } from '../logs';
+import { nativeBleManager } from './nativeBleManager';
+
+interface ConstructorParams extends AbstractApiConstructorParams {}
+
+const DEBUG_LOGS = true;
+
+const debugLog = (...args: any[]) => {
+    if (DEBUG_LOGS) {
+        // eslint-disable-next-line no-console
+        console.log('BleApi: ', ...args);
+    }
+    log(`BleApi: ${args.map(arg => JSON.stringify(arg)).join(' ')}`);
+};
+
+export class BleApi extends AbstractApi {
+    chunkSize = 244;
+    devices: BleDevice[] = [];
+
+    constructor({ logger }: ConstructorParams) {
+        super({ logger });
+
+        nativeBleManager.onconnect = async device => {
+            debugLog('onconnect', device.id);
+            const devicesDescriptors = await this.devicesToDescriptors();
+
+            this.emit('transport-interface-change', devicesDescriptors);
+        };
+
+        nativeBleManager.ondisconnect = async device => {
+            debugLog('ondisconnect', device.id);
+            this.emit('transport-interface-change', await this.devicesToDescriptors());
+        };
+    }
+
+    public async enumerate() {
+        debugLog('Transport enumeration');
+        try {
+            return this.success(await this.devicesToDescriptors());
+        } catch (err) {
+            debugLog('Enumerate error', err);
+
+            return this.unknownError(err, []);
+        }
+    }
+
+    private async devicesToDescriptors() {
+        const devices = await nativeBleManager.getAllConnectedDevices();
+
+        // TODO: implement proper device type
+        return devices.map(d => ({ path: d.id, type: DEVICE_TYPE.TypeT2 }));
```

Comment:
yes, we probably don't know the type yet? and also we maybe don't know the unique values that are use to match the type yet?

Reply:
it might be useful to add return value `DescriptorApiLevel` to this function. There were some additions to this type, you probably will need to stuff some dummy values (for example `product` field). they are not that much important for higher layers. But who knows, maybe you have some value like this also available.